pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -281,10 +281,14 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.CloudupMode
 
 	if ig.Spec.InstanceMetadata != nil && ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit != nil {
 		lt.HTTPPutResponseHopLimit = ig.Spec.InstanceMetadata.HTTPPutResponseHopLimit
+	} else if ig.IsControlPlane() && (b.Cluster.IsKubernetesLT("1.27") || !b.UseServiceAccountExternalPermissions()) {
+		lt.HTTPPutResponseHopLimit = fi.PtrTo[int64](3)
```

Comment:
Is there any reason we can't set this to 1 for k8s >= 1.27? Possibly in a follow-up PR to be done in 1.27.

Reply:
It is the condition from https://github.com/kubernetes/kops/pull/14572.

Reply:
I'm asking why we can't set it to 1 for control planes not using IRSA.

Reply:
https://github.com/kubernetes/kops/pull/10702 was the motivation for setting hop limit to 3 on control plane.

Reply:
I think it would be better to make such pods be host-network. I don't think this issue blocks this PR, but we should make a decision before we release 1.27.

Reply:
Ideally, we'd have some admission controller that ensures only kops-stuff can run on the control plane. But this is a very tricky thing to do. With PSS, it is at least easier to ensure only "privileged" namespaces can run with hostNetworking on, which in turn is a form of safety mechanism. So I am inclinded to agree with hostNetworking being better than max hops > 1.

Reply:
I agree too, just unrelated to this PR. The only purpose was to not pollute the IG config with these settings and hide them in `awsmodel` as defaults, not duplicating code.
Another topic is that it's already complicated to keep track of defaults or changes between versions. Would be nice to not have https://github.com/kubernetes/kops/pull/10702 in 1.26, this in 1.27 and so on...